ServerCore/Pages/Teams/Answers.cshtml.cs

```diff
@@ -35,43 +35,51 @@ public async Task OnGetAsync(SortOrder? sort, int teamId)
         {
             TeamID = teamId;
 
-            IQueryable<Submission> submissions = _context.Submissions
+            IQueryable<PuzzleStatePerTeam> puzzleStates = _context.PuzzleStatePerTeam
+                .Where((state) => state.TeamID == teamId && state.SolvedTime != null);
+
+            IQueryable<Submission> correctSubmissions = _context.Submissions
                 .Where((s) => s.TeamID == teamId && s.Response.IsSolution);
 
+            IQueryable<SubmissionView> finalSubmissions = puzzleStates.GroupJoin(
```

Comment:
The potential performance of this scares me, just because the query looks complex and I'm not confident that it's only making one server call. If you've measured the SQL traffic and are confident that this is a single call, OK.

But if not, then I feel like all you need is to evaluate puzzleStates.ToListAsync(), correctSubmissions.ToDictionaryAsync(s => s.PuzzleId), and then finalSubmissions is easy to build by enumerating puzzleStates and doing dictionary lookup in that dictionary.